runtime/src/main/java/org/corfudb/runtime/LogReplicationListener.java

```diff
@@ -0,0 +1,182 @@
+package org.corfudb.runtime;
+
+import com.google.common.base.Preconditions;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.LogReplication.ReplicationStatusVal;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.CorfuStreamEntries;
+import org.corfudb.runtime.collections.CorfuStreamEntry;
+import org.corfudb.runtime.collections.StreamListener;
+import org.corfudb.runtime.collections.TableSchema;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.view.Address;
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import static org.corfudb.runtime.LogReplicationUtils.REPLICATION_STATUS_TABLE;
+
+/**
+ * This is the interface that a client must subscribe to if it needs to observe and bifurcate the data updates received
+ * on Log Entry and Snapshot Sync.  The client's usecase is that it maintains a 'merged table' which contains data
+ * received through replication and local updates.  Log Replicator does not write to this merged table.  This
+ * listener will observe the writes and apply them to the merged table based on the client implementation.
+ *
+ *
+ * This interface sees ordered updates from :
+ * 1. client-streams from client-Namespace, and,
+ * 2. LrStatusTable from corfuSystem-Namespace.
+ *
+ * The client implementing this interface will only observe the data updates from client streams
```

Comment:
confused with this comment. Does the client receive updates from client streams only? or from both client and system namespace?

Reply:
Client will only see updates from client streams, subscribing system namespace is for internal handling.